src/nodeBuffer.ml

```diff
@@ -44,29 +45,341 @@ external allocFillStringWithEncoding: int -> fill:Js.String.t ->
 external allocFillBuffer: int -> fill:t -> t = "Buffer.alloc" [@@bs.val]
 external allocUnsafe: int -> t = "Buffer.allocUnsafe" [@@bs.val]
 external allocUnsafeSlow: int -> t = "Buffer.allocUnsafeSlow" [@@bs.val]
-external toString: t -> Js.String.t = "" [@@bs.send]
+
 external unsafeGet : t -> int -> int = "" [@@bs.get_index]
 external unsafeSet : t -> int -> int -> unit = "" [@@bs.set_index]
+
 external byteLengthString : Js.String.t -> int = "Buffer.byteLength" [@@bs.val]
 external byteLengthStringWithEncoding : Js.String.t -> encoding:Js.String.t -> int =
   "Buffer.byteLength" [@@bs.val]
 external byteLengthBuffer : t -> int = "Buffer.byteLength" [@@bs.val]
-external byteLengthInt8Array : int Int8Array.typed_array -> int =
+external byteLengthInt8Array : Int8Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint8Array : int Uint8Array.typed_array -> int =
+external byteLengthUint8Array : Uint8Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthInt16Array : int Int16Array.typed_array -> int =
+external byteLengthInt16Array : Int16Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint16Array : int Uint16Array.typed_array -> int =
+external byteLengthUint16Array : Uint16Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthInt32Array : int Int32Array.typed_array -> int =
+external byteLengthInt32Array : Int32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthUint32Array : int Uint32Array.typed_array -> int =
+external byteLengthUint32Array : Uint32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthFloat32Array : float Float32Array.typed_array -> int =
+external byteLengthFloat32Array : Float32Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
-external byteLengthFloat64Array : float Float64Array.typed_array -> int =
+external byteLengthFloat64Array : Float64Array.t -> int =
   "Buffer.byteLength" [@@bs.val]
 external byteLengthDataView : DataView.t -> int = "Buffer.byteLength" [@@bs.val]
 external byteLengthArrayBuffer : ArrayBuffer.t -> int = "Buffer.byteLength" [@@bs.val]
+
 external compare : t -> t -> int = "Buffer.compare" [@@bs.val]
+
+external concat : t array -> t = "Buffer.concat" [@@bs.val]
+external concatLength : t array -> length:int -> t = "Buffer.concat" [@@bs.val]
+
+external isEncoding : Js.String.t -> bool = "Buffer.isEncoding" [@@bs.val]
+
+type buffer
+external buffer : buffer = "Buffer" [@@bs.val]
+external poolSize : int = "Buffer.poolSize" [@@bs.val]
+external setPoolSize : buffer -> int -> int = "poolSize" [@@bs.set]
+let setPoolSize n = setPoolSize buffer n
+
+external copy : t -> t -> int = "" [@@bs.send]
+external copyOffset : t -> t -> targetStart:int -> int = "copy" [@@bs.send]
+external copyOffsetFromSourceOffset : t -> t -> targetStart:int -> sourceStart:int ->
```

Comment:
There's some inconcistencies here (and in the following functions). Maybe we should ditch `Source`, so it's just `FromOffset`, `FromRange`